### PR TITLE
Fix CUDA warnings

### LINKED ISSUE #1146: Deprecation warnings with cuSPARSE 11.8
Building ginkgo with CUDA 11.8 results in many warnings, mostly deprecation warnings from cuSPARSE. I'm attaching a complete build log of the [ginkgo-hpc-git AUR package](https://aur.archlinux.org/packages/ginkgo-hpc-git).

[ginkgo-hpc-git-r5521.8b036c64f4-1-x86_64-build.log](https://github.com/ginkgo-project/ginkgo/files/9792778/ginkgo-hpc-git-r5521.8b036c64f4-1-x86_64-build.log)

### PR DESCRIPTION
**Pull request by upsj**
Just a handful of deprecation warnings and unused variables

Fixes #1146

### CODE REVIEW

File: benchmark/utils/cuda_linops.cu [hunk 1]
```diff
@@ -349,6 +349,9 @@ private:
 #endif  // CUDA_VERSION < 11000
 
 
+#if CUDA_VERSION < 11030
```
Comment (tcojean) [inline benchmark/utils/cuda_linops.cu:352]: As far as I can see, the first deprecation here was 11.2.1: https://docs.nvidia.com/cuda/archive/11.2.1/cusparse/index.html#csrmvEx

File: benchmark/utils/cuda_linops.cu [hunk 2]
```diff
@@ -805,9 +815,13 @@ STUB_CREATE_SPARSELIB_LINOP(cusparse_csrmm);
     ((CUDA_VERSION >= 10020) && !(defined(_WIN32) || defined(__CYGWIN__)))
 IMPL_CREATE_SPARSELIB_LINOP(cusparse_gcsr,
                             detail::CusparseGenericCsr<etype, itype>);
-IMPL_CREATE_SPARSELIB_LINOP(
-    cusparse_gcsr2,
-    detail::CusparseGenericCsr<etype, itype, CUSPARSE_CSRMV_ALG2>);
+#if CUDA_VERSION >= 11031
```
Comment (tcojean) [inline benchmark/utils/cuda_linops.cu:818]: Same here for the `CSRMV` algorithm name deprecations

File: cuda/matrix/csr_kernels.cu [hunk 3]
```diff
@@ -363,7 +363,11 @@ bool try_general_sparselib_spmv(std::shared_ptr<const CudaExecutor> exec,
     if (b->get_stride() == 1 && c->get_stride() == 1) {
         auto vecb = cusparse::create_dnvec(b->get_size()[0], b_val);
         auto vecc = cusparse::create_dnvec(c->get_size()[0], c_val);
-        cusparseSpMVAlg_t alg = CUSPARSE_CSRMV_ALG1;
+#if CUDA_VERSION >= 11031
```
Comment (tcojean) [inline cuda/matrix/csr_kernels.cu:366]: same as earlier

**Review by tcojean — APPROVED** [review]
LGTM!

**Review by yhmtsai — APPROVED** [review]

File: benchmark/utils/cuda_linops.cu (hunk 1 above)
Comment (upsj) [reply benchmark/utils/cuda_linops.cu:352]: Thanks! I only checked major versions. Let me update that

**Pull request merged**